otherlibs/unix/Makefile

```diff
@@ -19,6 +19,9 @@ LIBNAME=unix
 
 EXTRACAMLFLAGS=-nolabels
 
+# dllunix.so particularly requires libm for modf symbols
+LDOPTS=$(NATIVECCLIBS)
```

Comment:
Why this is needed.
dllunix.so references modf which comes from libm, but the way it is built currently it only links to libc.so and libdl.so. On my linux system libc.so has a weak reference to modf and then (I guess) libm.so gets linked due to other reasons and symbol is resolved, but on android libc.so has no reference to modf and even though the binary links in libm.so - dllunix.so doesn't "see" it and cannot find modf symbol